spec/acceptance/reporting/event_spec.rb

```diff
@@ -23,34 +24,31 @@
     trigger_puppet_run(server, acceptable_exit_codes: [2])
     event = Helpers.get_single_record('em_event', query)
     additional_info = JSON.parse(event['additional_info'])
-
     expect(event['source']).to eql('Puppet')
     expect(event['type']).to eql('node_report_intentional_changes')
     expect(event['severity']).to eql('5')
     expect(event['message_key']).not_to be_empty
     expect(event['node']).not_to be_empty
-    expect(event['event_class']).to match(Regexp.new(Regexp.escape(server.uri)))
-    expect(event['description']).to match(Regexp.new(Regexp.escape(server.uri)))
+    expect(event['event_class']).to match(Regexp.new(Regexp.escape(server.run_shell('facter fqdn').stdout.chomp)))
+    expect(event['description']).to match(Regexp.new(Regexp.escape(server.run_shell('facter fqdn').stdout.chomp)))
```

Comment:
Doing run_shell twice is more expensive than it probably needs to be. Can we put this in a `let` or something so it only has to run it once to get the value?